app/controllers/vm_common.rb

```diff
@@ -573,15 +585,9 @@ def snap_vm
       @name = params[:name]
       @description = params[:description]
       if params[:name].blank? && !@record.try(:snapshot_name_optional?)
-        add_flash(_("Name is required"), :error)
-        @in_a_form = true
-        drop_breadcrumb(:name => _("Snapshot VM '%{name}'") % {:name => @record.name}, :url => "/vm_common/snap")
-        if session[:edit] && session[:edit][:explorer]
-          @edit = session[:edit]    # saving it to use in next transaction
-          javascript_flash(:spinner_off => true)
-        else
-          render :action => "snap"
-        end
+        render_missing_field(session, "Name")
+      elsif params[:description].blank? && @record.try(:snapshot_description_required?)
```

Comment:
minor, but I always prefer `try!` over `try`, cause the former one will raise an exception if you call it on nil.